cups/ipp-vars.c

```diff
@@ -221,8 +222,29 @@ _ippVarsSet(_ipp_vars_t *v,		/* I - IPP variables */
   if (!strcmp(name, "uri"))
   {
     char		uri[1024];	/* New printer URI */
+    char		resolved[1024];	/* Resolved mDNS URI */
+    char		value_uri[1024];	/* URI from value */
+    http_uri_status_t	uri_status;	/* URI status */
 
-    if (httpSeparateURI(HTTP_URI_CODING_ALL, value, v->scheme, sizeof(v->scheme), v->username, sizeof(v->username), v->host, sizeof(v->host), &(v->port), v->resource, sizeof(v->resource)) < HTTP_URI_STATUS_OK)
+    snprintf(value_uri, sizeof(value_uri), "%s", value);
```

Comment:
OK, so this copy is 100% unnecessary. Just call strstr on the original value string and then reassign value to the resolved URI.

Also, for future reference snprintf correctly adds a nul terminator at the end - you don't need to treat it like strncpy. Stylistically you should use "sizeof(foo) - 1" and not hardcode the last element, and for CUPS code you can use strlcpy (emulated when needed) which is a safe replacement for strncpy that makes sure the nul is on the end of the buffer. snprintf should only be used if you actually need to add something to the string.

Reply:
You're right, I'll update the code. Thank you for the tips too!